astronomer/providers/amazon/aws/sensors/emr.py

```diff
@@ -0,0 +1,51 @@
+from typing import Any, Dict
+
+from airflow import AirflowException
+from airflow.providers.amazon.aws.sensors.emr import EmrStepSensor
+
+from astronomer.providers.amazon.aws.triggers.emr import EmrStepSensorTrigger
+
+
+class EmrStepSensorAsync(EmrStepSensor):
+    """
+    Async (deferring) version of EmrStepSensor
+
+    Asks for the state of the step until it reaches any of the target states.
+    If it fails the sensor errors, failing the task.
+    With the default target states, sensor waits step to be completed.
+
+    :param job_flow_id: job_flow_id which contains the step check the state of
+    :param step_id: step to check the state of
+    :param target_states: the target states, sensor waits until
+        step reaches any of these states
+    :param failed_states: the failure states, sensor fails when
+        step reaches any of these states
+    """
+
+    def execute(self, context: Dict[str, Any]) -> None:
+        """Deferred and give control to trigger"""
+        self.defer(
+            timeout=self.execution_timeout,
+            trigger=EmrStepSensorTrigger(
+                job_flow_id=self.job_flow_id,
+                step_id=self.step_id,
+                target_states=self.target_states,
+                failed_states=self.failed_states,
+                aws_conn_id=self.aws_conn_id,
+                poke_interval=self.poke_interval,
+            ),
+            method_name="execute_complete",
+        )
+
+    def execute_complete(self, context: Dict[str, Any], event: Dict[str, Any]) -> None:
+        """
+        Callback for when the trigger fires - returns immediately.
+        Relies on trigger to throw an exception, otherwise it assumes execution was
+        successful.
+        """
+        if event:
+            if event["status"] == "error":
+                raise AirflowException(event["message"])
+
+            self.log.info("%s", event["message"])
+            self.log.info("%s completed successfully.", self.job_flow_id)
```

Comment:
Can we have f-strings here and across log statements in the PR too so that we have consistent string formatting functions used everywhere? Or is this a limitation on the base class logger that it does not accept f-strings?